metricq/history_client.py

```diff
@@ -30,8 +30,10 @@
 import uuid
 from asyncio import CancelledError, Task
 from asyncio.futures import Future
+from collections.abc import Iterable, Iterator
```

Comment:
Aren't the uses of `Iterator` actually a generator?

Reply:
Note that `Iterable` everywhere does not actually work since we do in fact use `next` sometimes. So I think `Iterator` in that case is the best compromise.